internal/gatewayapi/globalresources.go

```diff
@@ -18,7 +19,21 @@ import (
 const envoyTLSSecretName = "envoy"
 
 // ProcessGlobalResources processes global resources that are not tied to a specific listener or route
-func (t *Translator) ProcessGlobalResources(resources *resource.Resources, xdsIRs resource.XdsIRMap) error {
+func (t *Translator) ProcessGlobalResources(resources *resource.Resources, xdsIRs resource.XdsIRMap, gateways []*GatewayContext) error {
+	// Add the ProxyServiceCluster information for each gateway to the IR map
+	for _, gateway := range gateways {
```

Comment:
this should ideally iterate over infraIR, right now we iterate through all gateways even for mergeGateways case, can be handled in a follow up